certora/specs/RevertsMintBurnEthereum.spec

```
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+methods {
+    function owner() external returns address envfree;
+    function totalSupply() external returns uint256 envfree;
+    function balanceOf(address) external returns uint256 envfree;
+    function delegatee(address) external returns address envfree;
+    function delegatedVotingPower(address) external returns uint256 envfree;
+}
+
+// Check the revert conditions for the burn function.
+rule mintRevertConditions(env e, address to, uint256 amount) {
+    mathint totalSupplyBefore = totalSupply();
+    uint256 balanceOfSenderBefore = balanceOf(e.msg.sender);
+    uint256 toVotingPowerBefore = delegatedVotingPower(delegatee(to));
+
+    // Safe require as zero address can't possibly delegate voting power.
+    require delegatee(0) == 0;
+
+    // Safe require because if the delegatee is not zero the recipient's voting power excludes the newly minted value.
+    require delegatee(to) != 0 => toVotingPowerBefore <= totalSupply() - amount;
```

I don't get this one, why would the amount minted be less than the current total supply

And it seems like the following goes through:
```suggestion
    // Safe require because if the delegatee is not zero the recipient's voting power excludes the newly minted value.
    require delegatee(to) != 0 => toVotingPowerBefore <= totalSupply();
```

---

The rationale here was that to make the assumption that the newly minted amount wasn't in the total supply _before_ minting (that is before the token balance update checks, I could have used a variable `totalSupplyBefore` instead) but then I got confused and didn't realize that this `totalSupply` is already the value before minting.

---

Fixed in https://github.com/morpho-org/morpho-token-upgradeable/pull/98/commits/408a9357565e98c4ea6dd884a07e00044a798761.